atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/InitialBucketAssignerStateMachineBootstrapTask.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.sweep.asts.bucketingthings.BucketAssignerState;
+import com.palantir.atlasdb.sweep.asts.bucketingthings.SweepBucketAssignerStateMachineTable;
+import com.palantir.atlasdb.sweep.queue.ShardAndStrategy;
+import com.palantir.atlasdb.sweep.queue.ShardProgress;
+import com.palantir.atlasdb.sweep.queue.SweepQueueUtils;
+import com.palantir.atlasdb.table.description.SweeperStrategy;
+import java.util.Set;
+import java.util.function.LongSupplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class InitialBucketAssignerStateMachineBootstrapTask {
+    private final ShardProgress progress;
+    private final SweepBucketAssignerStateMachineTable stateMachineTable;
+    private final LongSupplier freshTimestampSupplier;
+
+    public InitialBucketAssignerStateMachineBootstrapTask(
+            ShardProgress progress,
+            SweepBucketAssignerStateMachineTable stateMachineTable,
+            LongSupplier freshTimestampSupplier) {
+        this.progress = progress;
+        this.stateMachineTable = stateMachineTable;
+        this.freshTimestampSupplier = freshTimestampSupplier;
+    }
+
+    public void run() {
+        if (stateMachineTable.doesStateMachineStateExist()) {
```

Comment:
In general for this class: do we want some logging to understand what the task is doing?